src/server/stream_family.cc

```diff
@@ -138,6 +150,321 @@ bool ParseRangeId(string_view id, RangeId* dest) {
   return ParseID(id, dest->exclude, 0, &dest->parsed_id);
 }
 
+/* This is a wrapper function for lpGet() to directly get an integer value
+ * from the listpack (that may store numbers as a string), converting
+ * the string if needed.
+ * The 'valid" argument is an optional output parameter to get an indication
+ * if the record was valid, when this parameter is NULL, the function will
+ * fail with an assertion. */
+static inline int64_t lpGetIntegerIfValid(unsigned char* ele, int* valid) {
+  int64_t v;
+  unsigned char* e = lpGet(ele, &v, NULL);
+  if (e == NULL) {
```

Comment:
If you're into C++-ifying this code, you could convert this (and below) to `nullptr` instead of `NULL` :)

Reply:
Well, there are many "opportunities" to convert this code to C++, so let me know if you think that's worthwhile. Otherwise I'll punt on them :)

Reply:
I just copied the code and did not bother. I would not write it like if writing it from scratch in c++

Reply:
Yeah, I figured that to be the case. Sure, SG.